### PR TITLE
3DTILES_implicit_tiling and 3DTILES_multiple_contents revisions

### PR DESCRIPTION
**Pull request by lilleyse**
I did a pass over `3DTILES_implicit_tiling` mainly to tighten up the introduction and remove redundancy in a few areas (including removing the glossary). I also added an example of implicit tiling's interaction with multiple contents.

Since we already have sections in other extensions showcasing `3DTILES_metadata` + `3DTILES_multiple_contents` and `3DTILES_implicit_tiling` + `3DTILES_multiple_contents` I pared down `3DTILES_multiple_contents` to the essentials. I may take another pass over it tomorrow to add back more details.

Previews

* [3DTILES_implicit_tiling](https://github.com/CesiumGS/3d-tiles/tree/implicit-revisions/extensions/3DTILES_implicit_tiling)
* [3DTILES_multiple_contents](https://github.com/CesiumGS/3d-tiles/tree/implicit-revisions/extensions/3DTILES_multiple_contents)

### CODE REVIEW

**Review by donmccurdy — COMMENTED** [review]
I agree with slimming down the `3DTILES_multiple_contents` spec, with one exception — it defines the schema for `subtree.3DTILES_multiple_contents.schema.json` but says nothing about it otherwise, which may be confusing. We could move that schema to implicit tiling? E.g.

```jsonc
  "tileAvailability": {"bufferView": 0},
  "childSubtreeAvailability": {"bufferView": 1},
  "contentAvailability": [
    {"bufferView": 2},
    {"bufferView": 3},
  ]
```

**Comment by lilleyse** [comment]
> We could move that schema to implicit tiling? E.g.
> 
> ```jsonc
>   "tileAvailability": {"bufferView": 0},
>   "childSubtreeAvailability": {"bufferView": 1},
>   "contentAvailability": [
>     {"bufferView": 2},
>     {"bufferView": 3},
>   ]
> ```

Yeah that could work. If a tileset doesn't use `3DTILES_multiple_contents` would `contentAvailability` be an array with a single element? That would be a breaking change but I think it's where we want to go. Once `3DTILES_multiple_contents` goes into baseline 3D Tiles `content` would likely become an array as well.

File: extensions/3DTILES_implicit_tiling/README.md [hunk 1]
```diff
@@ -336,21 +292,25 @@ Header fields:
...
+A **buffer view** is a contiguous subset of a buffer. A buffer view's `buffer` property is an integer index to identify the buffer. A buffer view has a `byteOffset` and a `byteLength` to describe the range of bytes within the buffer. The `byteLength` does not include any padding. Multiple buffer views may reference a single buffer.
 
 For efficient memory access, the `byteOffset` of a buffer view must be aligned to a multiple of 8 bytes.
 
+Tile availability, content availability, and child subtree availability may refer to either a buffer view containing an availability bitstream or a constant value (`1` meaning all elements are available, `0` meaning no elements are available).
+
+Availability bitstreams are packed in binary using the format described in the [Booleans](../../specification/Metadata#booleans) section of the Cesium 3D Metadata Specification.
```
[the remark this answers is not in the document]
Comment (lilleyse) [reply extensions/3DTILES_implicit_tiling/README.md]: Not intentional - thanks for pointing that out! Fixed

**Comment by donmccurdy** [comment]
> If a tileset doesn't use 3DTILES_multiple_contents would contentAvailability be an array with a single element?

I think that's the way to go, yes. 👍

**Comment by wallabyway** [comment]
I generate data-sets that use every layer and tile, so every tile is visible.  Is it ok, to not provide a 'tileAvailability' parameter?
Question:  Is 'tileAvailability' a requirement ?

**Comment by javagl** [comment]
> Question: Is 'tileAvailability' a requirement ?

@wallabyway  I'm not entirely sure whether this is what you referred to, but 

- The `tileAvailability` property is [required as of the schema](https://github.com/CesiumGS/3d-tiles/blob/main/extensions/3DTILES_implicit_tiling/schema/subtree/subtree.schema.json#L55)
- **But** if it is constant (i.e. when all tiles are available), then you can just say
  `"tileAvailability": { "constant": 1 }`
  as of [the availability spec](https://github.com/CesiumGS/3d-tiles/blob/main/extensions/3DTILES_implicit_tiling/schema/subtree/availability.schema.json#L23) (so you don't have to store a (huge) buffer that is filled with 1's...)

File: extensions/3DTILES_implicit_tiling/README.md [hunk 2]
```diff
@@ -47,42 +46,22 @@ This extension is required, meaning it must be placed in both the `extensionsUse
...
-For a complete list of terminology used, see the [Glossary](#glossary).
-
-## Use Cases
-
-_This section is non-normative_
-
-A key use for implicit tiling is enabling and/or accelerating tree traversal algorithms. Accessing a tile by coordinates is faster than traversing the entire tree. Likewise, raycasting algorithms and GIS algorithms can benefit from the abbreviated tree traversals. Tiles can be loaded immediately instead of going from top to bottom of a tree.
-
-Accessing tiles by coordinates also helps accelerate spatial queries. For example, the highest resolution tile that contains a given point can be quickly located by computing the coordinates of the tile directly. 
+The `3DTILES_implicit_tiling` extension may be added to any tile in the tileset. This extension defines how the tile is subdivided and where to locate content resources. The extension may be added to multiple tiles to create more complex subdivision schemes like double-headed quadtrees.
```
Comment (ptrgags) [inline extensions/3DTILES_implicit_tiling/README.md:60]: "double-headed quadtree" is jargony, if you're going to mention it, explain that it's a quadtree for each hemisphere (e.g. eastern and western), with another root tile that covers the whole world. A diagram would also help. Something like this: https://sandcastle.cesium.com/?src=Imagery%20Layers%20Manipulation.html

image
image

Except fix the numbering for the eastern hemisphere


Note that I have an example for this: https://github.com/CesiumGS/3d-tiles/issues/512#issuecomment-956636721 albeit this overview section might be too early on to show it.

**Review by ptrgags — COMMENTED** [review]
@lilleyse looks good, though if we want to talk about the double-headed quadtrees (and I think we should, it's pretty common), we should be more descriptive there.

File: extensions/3DTILES_implicit_tiling/README.md [hunk 3]
```diff
@@ -173,7 +150,7 @@ Implicit tiling only requires defining the subdivision scheme, refinement strate
 | --- | --- |
```
Comment (ptrgags) [inline extensions/3DTILES_implicit_tiling/README.md:150]: should we add a row in this table to say that `extensions` is constant for all descendant tiles? or should we not define this except in specific cases like for multiple contents?

**Comment by lilleyse** [comment]
If we introduce a breaking change for content availability we might also want to consider a breaking change for tile metadata where instead of using a `3DTILES_metadata` extension we add a `tileMetadata` property. This should help to further reduce schema inter-dependencies. Thoughts @donmccurdy?

<table>
<tr>
<td>Current&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td>
<td>Proposed&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td>
</tr>
<tr>
<td>

```json
{
  "buffers": [
    {
      "uri": "metadata.bin",
      "byteLength": 336
    }
  ],
  "bufferViews": [
    {
      "buffer": 0,
      "byteOffset": 0,
      "byteLength": 168
    },
    {
      "buffer": 0,
      "byteOffset": 168,
      "byteLength": 168
    }
  ],
  "tileAvailability": {
    "constant": 1
  },
  "contentAvailability": {
    "constant": 0
  },
  "childSubtreeAvailability": {
    "constant": 0
  },
  "extensions": {
    "3DTILES_metadata": {
      "class": "tile",
      "properties": {
        "minimumHeight": {
          "bufferView": 0
        },
        "maximumHeight": {
          "bufferView": 1
        }
      }
    }
  }
}
```

</td>
<td>

```json
{
  "buffers": [
    {
      "uri": "metadata.bin",
      "byteLength": 336
    }
  ],
  "bufferViews": [
    {
      "buffer": 0,
      "byteOffset": 0,
      "byteLength": 168
    },
    {
      "buffer": 0,
      "byteOffset": 168,
      "byteLength": 168
    }
  ],
  "tileAvailability": {
    "constant": 1
  },
  "contentAvailability": [{
    "constant": 0
  }],
  "childSubtreeAvailability": {
    "constant": 0
  },
  "tileMetadata": {
    "class": "tile",
    "properties": {
      "minimumHeight": {
        "bufferView": 0
      },
      "maximumHeight": {
        "bufferView": 1
      }
    }
  }
}
```

</td>
</tr>
</table>

**Comment by donmccurdy** [comment]
A couple changes that seem to follow, if we go that direction:

1. `3DTILES_metadata` extension must still be present on the tileset when using `tileMetadata`, to provide the schema.
2. `3DTILES_multiple_contents` could arguably go the same way, since "group" is not otherwise a metadata concept:

```jsonc
{
  "extensions": {
    "3DTILES_metadata": {
      "schema": {
        "classes": {
          "layer": {
            "properties": {
              "name": {"componentType": "STRING", "semantic": "NAME", "required": true},
              "color": {"type": "VEC3", "componentType": "UINT8"},
              "priority": {"componentType": "UINT32"}
            }
          }
        }
      }
    },
    "3DTILES_multiple_contents": { // ⚠️ changed from '3DTILES_metadata'
      "groupMetadata": { // ⚠️ changed from 'groups'
        "buildings": {
          "class": "layer",
          "properties": {
            "name": "Buildings Layer",
            "color": [128, 128, 128],
            "priority": 0
          }
        },
        "trees": {
          "class": "layer",
          "properties": {
            "name": "Trees Layer",
            "color": [10, 240, 30],
            "priority": 1
          }
        }
      }
    }
  },
  "root": {
    "extensions": {
      "3DTILES_multiple_contents": {
        "content": [
          // ⚠️ changed from '3DTILES_metadata'
          {"uri": "buildings.glb", "extensions": {"3DTILES_multiple_contents": {"group": "buildings"}}},
          {"uri": "trees.glb", "extensions": {"3DTILES_multiple_contents": {"group": "trees"}}}
        ]
      }
    },
    ...
  }
}
```

**Comment by lilleyse** [comment]
> 2. `3DTILES_multiple_contents` could arguably go the same way, since "group" is not otherwise a metadata concept:

While group metadata is most commonly used with `3DTILES_multiple_contents` it can also be used with single content tiles, and once multiple contents is core spec there would be no multiple contents extension to own the concept of group metadata. So I think group metadata still belongs in `3DTILES_metadata`.

But I also think there's too much nestedness with accessing group metadata - e.g. `root.extensions.3DTILES_multiple_contents.content[0].extensions.3DTILES_metadata.group` - and would love to simplify.

**Comment by lilleyse** [comment]
Most of the changes here have been incorporated in the latest revisions: https://github.com/CesiumGS/3d-tiles/tree/extension-revisions

**Pull request closed without merging**